lldb/source/Plugins/Protocol/MCP/Resource.cpp

```diff
@@ -0,0 +1,166 @@
+      if (!target_sp)
+        continue;
+      resources.emplace_back(GetTargetResource(i, j));
+    }
+  }
+
+  return resources;
+}
+
+llvm::Expected<protocol::ResourceResult>
+DebuggerResourceProvider::ReadResource(llvm::StringRef uri) const {
+  auto [protocol, path] = uri.split("://");
+
+  if (protocol != "lldb")
+    return createUnsupportedURIError(uri);
+
+  llvm::SmallVector<llvm::StringRef, 4> components;
+  path.split(components, '/');
+
+  if (components.size() < 2)
+    return createUnsupportedURIError(uri);
+
+  if (components[0] != "debugger")
+    return createUnsupportedURIError(uri);
+
+  lldb::user_id_t debugger_id;
+  if (components[1].getAsInteger(0, debugger_id))
+    return createStringError("invalid debugger id '{0}': {1}", components[1],
+                             path);
+
+  if (components.size() > 3) {
+    if (components[2] != "target")
+      return createUnsupportedURIError(uri);
+
+    lldb::user_id_t target_id;
+    if (components[3].getAsInteger(0, target_id))
+      return createStringError("invalid target id '{0}': {1}", components[3],
+                               path);
+
+    return ReadTargetResource(uri, debugger_id, target_id);
+  }
+
+  return ReadDebuggerResource(uri, debugger_id);
+}
+
+llvm::Expected<protocol::ResourceResult>
+DebuggerResourceProvider::ReadDebuggerResource(llvm::StringRef uri,
+                                               lldb::user_id_t debugger_id) {
+  lldb::DebuggerSP debugger_sp = Debugger::GetDebuggerAtIndex(debugger_id);
+  if (!debugger_sp)
+    return createStringError("invalid debugger id: {0}", debugger_id);
+
+  TargetList &target_list = debugger_sp->GetTargetList();
+  const size_t num_targets = target_list.GetNumTargets();
+
+  llvm::json::Value value = llvm::json::Object{{"debugger_id", debugger_id},
+                                               {"num_targets", num_targets}};
+
+  std::string json = llvm::formatv("{0}", value);
+
+  protocol::ResourceContents contents;
+  contents.uri = uri;
+  contents.mimeType = "application/json";
+  contents.text = json;
+
+  protocol::ResourceResult result;
+  result.contents.push_back(contents);
+  return result;
+}
+
+llvm::Expected<protocol::ResourceResult>
+DebuggerResourceProvider::ReadTargetResource(llvm::StringRef uri,
+                                             lldb::user_id_t debugger_id,
+                                             lldb::user_id_t target_id) {
+
+  lldb::DebuggerSP debugger_sp = Debugger::GetDebuggerAtIndex(debugger_id);
+  if (!debugger_sp)
+    return createStringError("invalid debugger id: {0}", debugger_id);
+
+  TargetList &target_list = debugger_sp->GetTargetList();
+  lldb::TargetSP target_sp = target_list.GetTargetAtIndex(target_id);
+  if (!target_sp)
+    return createStringError("invalid target id: {0}", target_id);
+
+  llvm::json::Object object{
+      {"debugger_id", debugger_id},
+      {"target_id", target_id},
+      {"arch", target_sp->GetArchitecture().GetTriple().str()}};
```

Comment:
Should we include the `platform` as well as the arch?

I think in the CLI the default description of a target is along the lines

```
(lldb) target list
Current targets:
* target #0: /path/to/exe ( arch=arm64-apple-macosx13.0.0, platform=host )
```

So, the platform seems like it might be worth including.